### PR TITLE
Practice Code Review - DO NOT MERGE!

### PR DESCRIPTION
**Pull request by renatodellosso**
This PR is to provide a practice code review for rookies. Your task is to find all the issues with the code (there are a lot).

### CODE REVIEW

**Review by Tr01ler — CHANGES_REQUESTED** [review]
Elevator motor main is not final

File: RobotCode/src/main/java/frc/robot/Ports.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+public class Ports
+{
+	public static final int CLAW_LEFT_CLOSE = 0;
+    public static final int CLAW_LEFT_OPEN = 0;
+    public static final int PNEUMATICS_HUB = 0;
+    public static final int CLAW_RIGHT_OPEN = 0;
+    public static final int CLAW_RIGHT_CLOSE = 0;
+    public static final int ELEVATOR_LIMIT_SWITCH = 0;
+    public static final AnalogInput ELEVATOR_POTENTIOMETER = null;
+    public static int ELEVATOR_MOTOR_MAIN = 0;
```
Comment (seankid) [inline RobotCode/src/main/java/frc/robot/Ports.java:14]: something is messed up here its either missing final designation or is incorrectly named

Comment (PPenguin1) [inline RobotCode/src/main/java/frc/robot/Ports.java:14]: elevatorMotorMain

File: RobotCode/src/main/java/frc/robot/RobotContainer.java [hunk 2]
```diff
@@ -12,7 +12,7 @@
 public class RobotContainer
 {
 
-	private final static ShuffleboardTab shuffleboard = Shuffleboard.getTab("Tab 1");
+	public final static ShuffleboardTab shuffleboard = Shuffleboard.getTab("Tab 1");
```
Comment (seankid) [inline RobotCode/src/main/java/frc/robot/RobotContainer.java:15]: idk something seems off here i cant place my finger on it

File: RobotCode/src/main/java/frc/robot/commands/MoveElevatorCommand.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.subsystems.ElevatorSubsystem;
+import java.util.function.DoubleSupplier;
+
+public class MoveElevatorCommand extends CommandBase
+{
+
+	public ElevatorSubsystem elevator;
+	public DoubleSupplier input;
```
Comment (PPenguin1) [inline RobotCode/src/main/java/frc/robot/commands/MoveElevatorCommand.java:11]: doubleSupplier

File: RobotCode/src/main/java/frc/robot/commands/MoveElevatorCommand.java [hunk 4]
```diff
@@ -0,0 +1,26 @@
+package frc.robot.commands;
+
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.subsystems.ElevatorSubsystem;
+import java.util.function.DoubleSupplier;
+
+public class MoveElevatorCommand extends CommandBase
+{
+
+	public ElevatorSubsystem elevator;
```
Comment (PPenguin1) [inline RobotCode/src/main/java/frc/robot/commands/MoveElevatorCommand.java:10]: elevatorSubsystem

File: RobotCode/src/main/java/frc/robot/subsystems/ClawIntakeSubsystem.java [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+import frc.lib.core.motors.TeamSparkMAX;
+import frc.robot.Ports;
+import frc.robot.RobotContainer;
+
+public class ClawIntakeSubsystem extends SubsystemBase
+{
+
+	public Compressor mainCompressor;
+
+	public DoubleSolenoid clawGrabberLeft, clawGrabberRight;
```
Comment (PPenguin1) [inline RobotCode/src/main/java/frc/robot/subsystems/ClawIntakeSubsystem.java:17]: doubleSolenoid

File: RobotCode/src/main/java/frc/robot/subsystems/ElevatorSubsystem.java [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+import frc.robot.RobotContainer;
+import frc.robot.commands.MoveElevatorCommand;
+
+public class ElevatorSubsystem extends SubsystemBase
+{
+
+	static ITeamTalon elevatorMotorMain, elevatorMotorSub;
+	static double targetPosition = Constants.MINIMUM_ELEVATOR_POSITION;
+	static final double DEADBAND_VALUE = Constants.ELEVATOR_DEADBAND_VALUE;
+	static AnalogPotentiometer potentiometer;
```
Comment (PPenguin1) [inline RobotCode/src/main/java/frc/robot/subsystems/ElevatorSubsystem.java:26]: analogPotentiometer

File: RobotCode/src/main/java/frc/robot/subsystems/ElevatorSubsystem.java [hunk 7]
```diff
@@ -0,0 +1,170 @@
...
+	static ClawIntakeSubsystem intake;
```
Comment (PPenguin1) [inline RobotCode/src/main/java/frc/robot/subsystems/ElevatorSubsystem.java:27]: clawIntakeSubsystem

**Review by PPenguin1 — COMMENTED** [review]
Public static double ELEVATOR_DEADBAND_VALUE to elevatorDeadbandValue

File: RobotCode/src/main/java/frc/robot/constants/Constants.java [hunk 8]
```diff
@@ -3,4 +3,12 @@
 public class Constants
 {
 
+	public static final double MINIMUM_ELEVATOR_POSITION = 0;
+	public static double ELEVATOR_DEADBAND_VALUE = 0;
+	public static final int maxElevatorMotorSpeed = 0;
+	public static final float ELEVATOR_POWER_EXPONENT = 0;
+	public final int ELEVATOR_MAX_POWER_CHANGE = 0;
```
Comment (seankid) [inline RobotCode/src/main/java/frc/robot/constants/Constants.java:10]: Should be named differently

**Review by seankid — COMMENTED** [review]
several things are mid

File: RobotCode/src/main/java/frc/robot/subsystems/ElevatorSubsystem.java [hunk 9]
```diff
@@ -0,0 +1,170 @@
...
+		speed *= Constants.maxElevatorMotorSpeed;
+
+		speed = getCappedPower(speed);
+
+		double currentPower = elevatorMotorMain.get();
+
+		if ((speed < currentPower))
+		{
+			newPower = Math.max(speed, currentPower - Constants.ELEVATOR_MAX_POWER_CHANGE);
+		} // /jack morton lawrence torrance jack morton lawrence torrance jack morton
```
Comment (li46554) [inline RobotCode/src/main/java/frc/robot/subsystems/ElevatorSubsystem.java:101]: silly comment

**Pull request closed without merging**